docs/spec.md

```diff
@@ -2633,6 +2633,54 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
```

Comment:
If the corresponding parts of the ODS and StablehloOps.cpp match the spec, check that the "Verification" and "Type Inference" columns in [status.md](https://github.com/openxla/stablehlo/blob/main/docs/status.md) say "yes".